packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -238,6 +268,16 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
   @override
   Widget buildToolbar(BuildContext context, Rect globalEditableRegion, Offset position, TextSelectionDelegate delegate) {
     assert(debugCheckHasMediaQuery(context));
+
+    final _ArrowDirection direction =
+    (globalEditableRegion.top > _kToolbarArrowInvertDistance)
```

Comment:
So here instead of checking against a const value, I'd check against media query padding.